assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -1668,5 +1668,32 @@ describe("DiversionPage", () => {
         ).toBeChecked()
       })
     })
+
+    test("when no route is selected, clicking the finish button shows error message", async () => {
+      const routes = routeFactory.buildList(2)
+
+      render(
+        <RoutesProvider routes={routes}>
+          <DiversionPage
+            originalRoute={{
+              routePattern: undefined,
+              route: undefined,
+            }}
+          />
+        </RoutesProvider>
+      )
+
+      expect(
+        screen.getByRole("combobox", { name: "Choose route" })
+      ).not.toHaveAccessibleErrorMessage()
+
+      await userEvent.click(
+        screen.getByRole("button", { name: "Start drawing detour" })
+      )
+
+      expect(
+        screen.getByRole("combobox", { name: "Choose route" })
+      ).toHaveAccessibleErrorMessage("Select a route to continue.")
```

Comment:
note: I would've liked to use `toBeVisible` in these tests, as that's more accurate to what we expect, but because we don't have CSS in our tests, we don't have the ability to check that because bootstrap styles the visibility entirely with the presence of a class on the `<Form>` element, we could _manually_ add the `hidden` prop on the error container though, which would be a stopgap until we could get our tests to check CSS as well